src/containers/consignment.rs

```diff
@@ -355,8 +356,23 @@ impl<const TRANSFER: bool> Consignment<TRANSFER> {
         if self.transfer != TRANSFER {
             status.add_warning(Warning::Custom(s!("invalid consignment type")));
         }
-        // TODO: check that interface ids match implementations
-        // TODO: check bundle ids listed in terminals are present in the consignment
+        // check ifaceid match implementation
+        for (iface, iimpl) in self.ifaces.iter() {
+            if iface.iface_id() != iimpl.iface_id {
+                status.add_warning(Warning::Custom(s!(
+                    "interface ids doesn't match implementations"
+                )));
```

Comment:
This should help with debugging:
```suggestion
                status.add_warning(Warning::Custom(format!(
                    "implementation {} targets different interface {} than expected {}", iimpl.impl_id(), iimpl.iface_id, iface.iface_id()
                )));
```